### ISSUE TITLE
clippy::create_dir should not remove `fs::` path qualifier

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#14994]
Using the following flags

`--force-warn clippy::create-dir`

this code:

```rust
use std::fs;

fn main() -> std::io::Result<()> {
    fs::create_dir("/some/dir")?;
    Ok(())
}

```

caused the following diagnostics:

```
    Checking _main v0.1.0 (/tmp/icemaker_global_tempdir.OL3SSPkKyamm/icemaker_clippyfix_tempdir.r8BjAO9ZzJZZ/_main)
warning: calling `std::fs::create_dir` where there may be a better way
 --> src/main.rs:4:5
  |
4 |     fs::create_dir("/some/dir")?;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#create_dir
  = note: requested on the command line with `--force-warn clippy::create-dir`
help: consider calling `std::fs::create_dir_all` instead
  |
4 -     fs::create_dir("/some/dir")?;
4 +     create_dir_all("/some/dir")?;
  |

warning: `_main` (bin "_main") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.11s

```

However after applying these diagnostics, the resulting code:

```rust
use std::fs;

fn main() -> std::io::Result<()> {
    create_dir_all("/some/dir")?;
    Ok(())
}

```
no longer compiled:

```
    Checking _main v0.1.0 (/tmp/icemaker_global_tempdir.OL3SSPkKyamm/icemaker_clippyfix_tempdir.r8BjAO9ZzJZZ/_main)
warning: error applying suggestions to `src/main.rs`

The full error message was:

> cannot replace slice of data that was already replaced

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

warning: failed to automatically apply fixes suggested by rustc to crate `_main`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0425]: cannot find function `create_dir_all` in this scope
 --> src/main.rs:4:5
  |
4 |     create_dir_all("/some/dir")?;
  |     ^^^^^^^^^^^^^^ not found in this scope
  |
help: consider importing one of these functions
  |
1 + use crate::fs::create_dir_all;
  |
1 + use std::fs::create_dir_all;
  |

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0425`.
Original diagnostics will follow.

error[E0425]: cannot find function `create_dir_all` in this scope
 --> src/main.rs:4:5
  |
4 |     create_dir_all("/some/dir")?;
  |     ^^^^^^^^^^^^^^ not found in this scope
  |
help: consider importing one of these functions
  |
1 + use crate::fs::create_dir_all;
  |
1 + use std::fs::create_dir_all;
  |

For more information about this error, try `rustc --explain E0425`.
error: could not compile `_main` (bin "_main") due to 1 previous error
warning: build failed, waiting for other jobs to finish...
warning: error applying suggestions to `src/main.rs`

The full error message was:

> cannot replace slice of data that was already replaced

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

warning: failed to automatically apply fixes suggested by rustc to crate `_main`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0425]: cannot find function `create_dir_all` in this scope
 --> src/main.rs:4:5
  |
4 |     create_dir_all("/some/dir")?;
  |     ^^^^^^^^^^^^^^ not found in this scope
  |
help: consider importing one of these functions
  |
1 + use crate::fs::create_dir_all;
  |
1 + use std::fs::create_dir_all;
  |

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0425`.
Original diagnostics will follow.

error: could not compile `_main` (bin "_main" test) due to 1 previous error

```

Version:
```
rustc 1.89.0-nightly (cf423712b 2025-06-05)
binary: rustc
commit-hash: cf423712b9e95e9f6ec84b1ecb3d125e55ac8d56
commit-date: 2025-06-05
host: x86_64-unknown-linux-gnu
release: 1.89.0-nightly
LLVM version: 20.1.5

```
